src/DevTools/__test__/MockRelayRenderer.test.tsx

```diff
@@ -0,0 +1,193 @@
+import { renderWithLoadProgress } from "Artsy/Relay/renderWithLoadProgress"
+import { ContextConsumer } from "Artsy/SystemContext"
+import cheerio from "cheerio"
+import { mount, render } from "enzyme"
+import * as React from "react"
+import { createFragmentContainer, graphql, QueryRenderer } from "react-relay"
+import { MockRelayRenderer, renderRelayTree } from "../MockRelayRenderer"
+
+jest.unmock("react-relay")
+
+const Metadata = createFragmentContainer(
+  (props: { artworkMetadata: MockRelayRenderer_artworkMetadata }) => (
+    <div>{props.artworkMetadata.title}</div>
+  ),
+  graphql`
+    fragment MockRelayRenderer_artworkMetadata on Artwork {
+      title
+    }
+  `
+)
+
+const Artwork = createFragmentContainer(
+  (props: { artwork: MockRelayRenderer_artwork }) => (
+    <div>
+      <img src={props.artwork.image.url} />
+      <Metadata artworkMetadata={props.artwork} />
+      {props.artwork.artist && (
+        <ArtistQueryRenderer id={props.artwork.artist.id} />
+      )}
+    </div>
+  ),
+  graphql`
+    fragment MockRelayRenderer_artwork on Artwork {
+      image {
+        url
+      }
+      artist {
+        id
+      }
+      ...MockRelayRenderer_artworkMetadata
+    }
+  `
+)
+
+const Artist = createFragmentContainer(
+  (props: { artist: MockRelayRenderer_artist }) => (
+    <div>{props.artist.name}</div>
+  ),
+  graphql`
+    fragment MockRelayRenderer_artist on Artist {
+      name
+    }
+  `
+)
+
+const ArtistQueryRenderer = (props: { id: string }) => (
+  <ContextConsumer>
+    {({ relayEnvironment }) => {
+      return (
+        <QueryRenderer<MockRelayRendererArtistQuery>
+          environment={relayEnvironment}
+          variables={props}
+          query={graphql`
+            query MockRelayRendererArtistQuery($id: String!) {
+              artist(id: $id) {
+                ...MockRelayRenderer_artist
+              }
+            }
+          `}
+          render={renderWithLoadProgress(Artist)}
+        />
+      )
+    }}
+  </ContextConsumer>
+)
+
+const query = graphql`
+  query MockRelayRendererQuery {
+    artwork(id: "mona-lisa") {
+      ...MockRelayRenderer_artwork
+    }
+  }
+`
+
+function renderToString(element: React.ReactElement<any>) {
+  return cheerio.html(render(element))
+}
+
+describe("MockRelayRenderer", () => {
+  it("renders a Relay tree", done => {
+    const tree = mount(
+      <MockRelayRenderer
+        Component={Artwork}
+        query={query}
+        mockResolvers={{
+          Artwork: () => ({
+            title: "Mona Lisa",
+            image: {
+              url: "http://test/image.jpg",
+            },
+            artist: null,
+          }),
+        }}
+      />
+    )
+    setTimeout(() => {
+      expect(tree.html()).toEqual(
+        renderToString(
+          <div>
+            <img src="http://test/image.jpg" />
+            <div>Mona Lisa</div>
+          </div>
+        )
+      )
+      done()
+    }, 10)
```

Comment:
Will this ever unexpectedly fail in the future?

Reply:
Timers aren’t mocked by default: https://jestjs.io/docs/en/timer-mocks.html

What part of he code are you thinking of? I don’t really see a reason for this to fail (other than the code being broken).

Reply:
Ah, was thinking about `jest.useFakeTimers()` and the seemingly arbitrary `10ms` use here.

Reply:
Yeah that value is arbitrary, but shouldn’t break unless it is brought down to a very low value.